content/en/blog/_posts/2025-XX-XX-nftables-kube-proxy/index.md

```diff
@@ -0,0 +1,243 @@
+---
+layout: blog
+title: "NFTables mode for kube-proxy"
+date: 2024-08-21
+slug: nftables-kube-proxy
+author: >
+  Dan Winship (Red Hat)
+---
+
+A new nftables mode for kube-proxy was introduced as an alpha feature
+in Kubernetes 1.29. Currently in beta, it will officially be GA as of
+1.33. The new mode fixes long-standing performance problems with the
+iptables mode and is now recommended for most users running on systems
+with reasonably-recent kernels. (For compatibility reasons, even once
+nftables becomes GA, iptables will still be the _default_.)
+
+## Why NFTables? Part 1: Data Plane Latency
+
+The iptables API was designed for implementing simple firewalls, and
+has problems scaling up to support service proxying in a large
+Kubernetes cluster with tens of thousands of services.
+
+In general, the ruleset generated by kube-proxy in iptables mode has a
+number of iptables rules proportional to the sum of the number of
+services and the total number of endpoints. In particular, at the "top
+level" of the ruleset, there is one rule to test each possible service
+IP (and port) that a packet might be addressed to:
+
+```
+# If the packet is addressed to 172.30.0.41:80, then jump to the chain
+# KUBE-SVC-XPGD46QRK7WJZT7O for further processing
+-A KUBE-SERVICES -m comment --comment "namespace1/service1:p80 cluster IP" -m tcp -p tcp -d 172.30.0.41 --dport 80 -j KUBE-SVC-XPGD46QRK7WJZT7O
+
+# If the packet is addressed to 172.30.0.42:443, then...
+-A KUBE-SERVICES -m comment --comment "namespace2/service2:p443 cluster IP" -m tcp -p tcp -d 172.30.0.42 --dport 443 -j KUBE-SVC-GNZBNJ2PO5MGZ6GT
+
+# etc...
+-A KUBE-SERVICES -m comment --comment "namespace3/service3:p80 cluster IP" -m tcp -p tcp -d 172.30.0.43 --dport 80 -j KUBE-SVC-X27LE4BHSL4DOUIK
+```
+
+This means that when a packet comes in, the time it takes the kernel
+to check it against all of the service rules is **O(n)** in the number
+of services. As the number of services increases, both the average and
+the worst-case latency for the first packet of a new connection
+increases (with the difference between "best-case", "average", and
+"worst-case" being mostly determined by whether a given service IP
```

Comment:
```suggestion
increases (with the difference between best-case, average, and
worst-case being mostly determined by whether a given service IP address
```

Reply:
OK, so: explain that in parens, maybe? The quotes don't convey that.

You could use italics if you prefer; try not to use quoting.